### PR TITLE
doc: add types to template

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
@DanielSidhion here's a proposal, following up on https://github.com/NixOS/nixpkgs/pull/281211#issuecomment-1892775806

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by DanielSidhion — APPROVED** [review]
Perhaps it would be useful to mention the names from https://nixos.org/manual/nix/stable/language/values (or some other resource), just to try to get people use the same names, e.g. "Attribute set" instead of "Attrset" (or something else).

This is optional though, because imo it's more important to have *some* type even if not consistent throughout the docs (which we can be fixed later, especially because the definition list format is already well-defined).

**Comment by fricklerhandwerk** [comment]
Good point, added a reference to types.

**Review by DanielSidhion — APPROVED** [review]

File: doc/README.md [hunk 1]
```diff
@@ -188,24 +188,24 @@ In that case, please open an issue about the particular documentation convention
...
   Description of what `coolFunction` does.
   `coolFunction` expects a single argument which should be an attribute set, with the following possible attributes:
 
-  `name`
+  `name` (String)
 
   : The name of the resulting image.
 
-  `tag` _optional_
+  `tag` (String, _optional_)
```
Comment (DanielSidhion) [inline doc/README.md:204]:
```suggestion
  `tag` (String; _optional_)
```

When there's more than one type allowed, sometimes I have to use a comma to separate them, so I started using a semicolon to separate the types from additional properties. Example: `Path, List of Paths, or Null; _optional_`.

Also open to suggestions on a different approach.

Comment (fricklerhandwerk) [reply doc/README.md:204]: "Around here" we lean towards the Haskell type notation e.g. `(String | Integer)`, but we probably shouldn't use it without explanation. I've seen multiple exchanges where people discussed which notation to use and how to introduce it to readers and maintainers. From experience, I suggest not thinking too hard about it in one place, as that will just prevent any user-facing progress at all, and instead evolve it incrementally.

**Pull request merged**